xmppserver/src/test/java/org/jivesoftware/util/cache/DefaultExternalizableUtilTest.java

```diff
@@ -0,0 +1,622 @@
+     * implementation results in a value that is equal to the original input.
+     */
+    // TODO @Test
+    public void testWriteReadSerializableMap() throws Exception
+    {
+        // TODO implement me (and variants for collections that are empty or contain null values).
+    }
+
+    /**
+     * Asserts that the process of writing and reading a Collection of Strings through the {@link DefaultExternalizableUtil}
+     * implementation results in a value that is equal to the original input.
+     */
+    @Test
+    public void testWriteReadStringCollection() throws Exception
+    {
+        // TODO implement me (and variants for collections that are empty or contain null values).
+    }
+
+    /**
+     * Checks if two maps are equal.
+     *
+     * @param first A map
+     * @param second Another map
+     * @return if both maps are equal, otherwise false.
+     */
+    private static boolean areEqualString(Map<String, String> first, Map<String, String> second) {
+        if (first.size() != second.size()) {
+            return false;
+        }
+
+        return first.entrySet().stream()
+            .allMatch(e -> (e.getValue() == null && second.get(e.getKey()) == null)
+                        || (e.getValue().equals(second.get(e.getKey()))) );
+    }
+
+    /**
+     * Checks if two maps are equal.
+     *
+     * @param first A map
+     * @param second Another map
+     * @return if both maps are equal, otherwise false.
+     */
+    private static boolean areEqualStrings(Map<String, Set<String>> first, Map<String, Set<String>> second) {
+        if (first.size() != second.size()) {
+            return false;
+        }
+
+        return first.entrySet().stream()
+            .allMatch(e -> (e.getValue() == null && second.get(e.getKey()) == null)
+                || (e.getValue().equals(second.get(e.getKey()))) );
+    }
+
+    /**
+     * Checks if two maps are equal.
+     *
+     * @param first A map
+     * @param second Another map
+     * @return if both maps are equal, otherwise false.
+     */
+    private static boolean areEqualLongInt(Map<Long, Integer> first, Map<Long, Integer> second) {
+        if (first.size() != second.size()) {
+            return false;
+        }
+
+        return first.entrySet().stream()
+            .allMatch(e -> (e.getValue() == null && second.get(e.getKey()) == null)
+                || (e.getValue().equals(second.get(e.getKey()))) );
+    }
+
+    /**
+     * Utility class that generates a DataOutput and DataInput that share a common resource, meaning that what is
+     * written to the DataOutput becomes available for reading on the DataInput.
+     *
+     * Instances are single-use. DataOutput should be written to before DataInput is obtained (this implementation was
+     * designed for unit testing only, not for re-use elsewhere).
+     */
+    public static class Bus
```

Comment:
Perhaps make the class/constructor private to re-enforce the "for testing only"

Reply:
Well, it's only available in the test scope anyway. This way, we can re-abuse it in other tests, if we need to.